src/nvim/terminal.c

```diff
@@ -217,10 +214,14 @@ Terminal *terminal_open(TerminalOptions opts)
   vterm_set_utf8(rv->vt, 1);
   // Setup state
   VTermState *state = vterm_obtain_state(rv->vt);
+  // Only relevant in terminal true color mode
+  if (true_color) {
+    vterm_state_set_bold_highbright(state, true);
```

Comment:
From reading the code I didn't get the idea that the `g:terminal` options were used only in true color mode (they supported 256 colors). I personally wasn't using but it's something to take into account. 
An option `g:terminal_bold_highbright` anyway is needed, and I wouldn't put it behind a true color mode check since it would be relevant in both cases.